src/data/history.ts

```diff
@@ -419,17 +419,41 @@ const BLANK_UNIT = " ";
 export const computeHistory = (
   hass: HomeAssistant,
   stateHistory: HistoryStates,
+  entityIds: string[],
   localize: LocalizeFunc,
   sensorNumericalDeviceClasses: string[],
   splitDeviceClasses = false
 ): HistoryResult => {
   const lineChartDevices: { [unit: string]: HistoryStates } = {};
   const timelineDevices: TimelineEntity[] = [];
-  if (!stateHistory) {
+
+  const localStateHistory: HistoryStates = {};
+
+  // Create a limited history from stateObj if entity has no recorded history.
+  entityIds.forEach((entity) => {
+    if (entity in stateHistory) {
+      localStateHistory[entity] = stateHistory[entity];
+    } else if (hass.states[entity]) {
+      localStateHistory[entity] = [
+        {
+          s: hass.states[entity].state,
+          a: hass.states[entity].attributes,
+          lu: new Date(hass.states[entity].last_updated).getTime() / 1000,
```

Comment:
I think this should be `last_changed` and not `last_updated`?

Reply:
I think I convinced myself that last_updated would be more correct, in case we were displaying a chart with attributes (like climate). 

If we use last_changed, then we would render that the attributes were constant all the way back to the time of the last state change, but that would be possibly innacurate, as the attributes may have been changing after that point.

If we cutoff the graph at the time of last attribute change, then we display potentially less data, but guarantee that what is shown is actually correct.